tests/std/tests/P2286R8_text_formatting_debug_enabled_specializations/test.cpp

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <cstddef>
+#include <format>
+#include <memory_resource>
+#include <string>
+#include <string_view>
+#include <type_traits>
+#include <utility>
+
+#include <test_format_support.hpp>
+
+#define STR(Str) TYPED_LITERAL(CharT, Str)
+
+using namespace std;
+
+template <class F>
+concept DebugEnabledSpecialization = is_default_constructible_v<F> && requires(F& fmt) {
+    { fmt.set_debug_format() } noexcept -> same_as<void>;
+};
+
+template <class CharT>
+consteval bool check_debug_enabled_specializations() {
+    static_assert(DebugEnabledSpecialization<formatter<char, CharT>>);
+    static_assert(DebugEnabledSpecialization<formatter<CharT, CharT>>);
+    static_assert(DebugEnabledSpecialization<formatter<CharT*, CharT>>);
+    static_assert(DebugEnabledSpecialization<formatter<const CharT*, CharT>>);
+    static_assert(DebugEnabledSpecialization<formatter<CharT[3], CharT>>);
+    static_assert(DebugEnabledSpecialization<formatter<basic_string<CharT>, CharT>>);
+    static_assert(DebugEnabledSpecialization<formatter<pmr::basic_string<CharT>, CharT>>);
+    static_assert(DebugEnabledSpecialization<formatter<basic_string_view<CharT>, CharT>>);
+
+    static_assert(!DebugEnabledSpecialization<formatter<signed char, CharT>>);
+    static_assert(!DebugEnabledSpecialization<formatter<short, CharT>>);
+    static_assert(!DebugEnabledSpecialization<formatter<int, CharT>>);
+    static_assert(!DebugEnabledSpecialization<formatter<long, CharT>>);
+    static_assert(!DebugEnabledSpecialization<formatter<long long, CharT>>);
+
+    static_assert(!DebugEnabledSpecialization<formatter<unsigned char, CharT>>);
+    // NB: formatter<unsigned short, CharT> is special case, see below
+    static_assert(!DebugEnabledSpecialization<formatter<unsigned int, CharT>>);
+    static_assert(!DebugEnabledSpecialization<formatter<unsigned long, CharT>>);
+    static_assert(!DebugEnabledSpecialization<formatter<unsigned long long, CharT>>);
+
+    static_assert(!DebugEnabledSpecialization<formatter<bool, CharT>>);
+    static_assert(!DebugEnabledSpecialization<formatter<nullptr_t, CharT>>);
+    static_assert(!DebugEnabledSpecialization<formatter<void*, CharT>>);
+    static_assert(!DebugEnabledSpecialization<formatter<const void*, CharT>>);
+
+    // NB: wchar_t might be defined as a typedef for unsigned short (with '/Zc:wchar_t-')
+    static_assert(!DebugEnabledSpecialization<formatter<unsigned short, CharT>>
+                  == (same_as<CharT, char> || !same_as<CharT, unsigned short>) );

```

Comment:
This can be simplified. `same_as<CharT, char>` implies `!same_as<CharT, unsigned short>` so we can just test the latter. Then this is testing `!X == !Y` so we can simplify again.